server/db/driver/pg/orders.go

```diff
@@ -268,13 +261,40 @@ func (a *Archiver) CancelOrder(lo *order.LimitOrder) error {
 	return a.updateOrderStatus(lo, orderStatusCanceled)
 }
 
-// RevokeOrder updates a LimitOrder with revoked status, which is used for
+// RevokeOrder updates an Order with revoked status, which is used for
 // DEX-revoked orders rather than orders matched with a user's CancelOrder. If
 // the order does not exist in the Archiver, RevokeOrder returns
-// ErrUnknownOrder. To store a new limit order with revoked status, use
-// StoreOrder.
-func (a *Archiver) RevokeOrder(lo *order.LimitOrder) error {
-	return a.updateOrderStatus(lo, orderStatusRevoked)
+// ErrUnknownOrder. This may change orders with status executed to revoked,
+// which may be unexpected.
+func (a *Archiver) RevokeOrder(ord order.Order) (cancelID order.OrderID, timeStamp time.Time, err error) {
+	// Revoke the targeted order.
+	err = a.updateOrderStatus(ord, orderStatusRevoked)
+	if err != nil {
+		return
+	}
+
+	// Create a server-generated cancel order to record this action.
+	timeStamp = time.Now().Truncate(time.Millisecond).UTC()
+	var commit order.Commitment
+	rand.Read(commit[:])
```

Comment:
Initially I wanted to store such server-generated cancel orders with a zero-value commit (or maybe a NULL in the table), but order.ValidateOrder and the UNIQUE constraint of the commit table column make this harder to do.  It's not impossible, but the easiest thing to do right now is to generate a random and unused commitment for which no preimage exists.